### PR TITLE
Merge GeoffreyBooth proposal into README

### PR DESCRIPTION
**Pull request by GeoffreyBooth**
@rattrayalex I wanted to submit this as a PR rather than just overwriting the README, since I changed or removed some of the text you had written in the README. Please merge in if my edits look good to you.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,62 +22,102 @@
...
-- [Elm](http://elm-lang.org/) – Haskell-based, only works in the browser.
-- [Frappe](https://github.com/lydell/frappe) (just an idea)
-- [TacoScript](https://github.com/forivall/tacoscript) (in the early stages)
+Besides the [leadership](https://github.com/coffeescript6/discuss/issues/3) and [compiler starting point](https://github.com/coffeescript6/discuss/issues/25) issues mentioned above, the [Issues](https://github.com/coffeescript6/discuss/issues) for this repo covers the various ES2015+ features and discussions regarding how to implement them. If you don’t see your favorite ES2015+ feature there, or want to propose your own new feature for the language, please [open an issue](https://github.com/coffeescript6/discuss/issues/new). That’s also the place to discuss the specifics of how to move this project forward.
 
+If you disagree with any part of this document, please [open a pull request](https://github.com/coffeescript6/discuss/pulls) with a suggested revision and we can discuss it.
 
 ## Relevant Discussions
 - https://github.com/jashkenas/coffeescript/issues/4078
```
Comment (mrmowgli) [inline README.md]: Looks great to me, seems to have everything in there! +1

File: README.md [hunk 2]
```diff
@@ -21,62 +22,102 @@
...
+### Modernizing Output
 
+Lastly, CoffeeScript’s output should be modernized as much as possible. Fat arrows should output as fat arrows, destructuring should output as destructuring, etc. But this should be the **last priority**—it makes our source code cleaner, and debugging eventually easier when ES2015 features are supported natively in runtimes, but otherwise ES2015 output has little benefit for developers. That said, *new* features, especially ES2015+ ones, should definitely be output as ES2015+ code.
 
-## Possible Alternatives
-There are a number of interesting new languages that have sprung up, some of them listed below. None seem to capture the essence of "unfancy JavaScript" accomplished in CoffeeScript, but may be sources of inspiration nonetheless:
+## A New Compiler
+
+The current CoffeeScript 1.x compiler is extremely well commented, but it really needs those comments. Understanding it will bend your brain into knots. The original goal of [CoffeeScript Redux](https://github.com/michaelficarra/CoffeeScriptRedux) was to refactor CoffeeScript’s codebase, not to add new features.
```
Comment (lydell) [inline README.md]: Redux is not a refactor, it’s a complete rewrite.

File: README.md [hunk 3]
```diff
@@ -21,62 +22,102 @@
...
+
+# Proposal for the Future of CoffeeScript
 
 ## Goals
 
-One clear goal of CS6 would be adoption of ES6+ features, either overwriting CoffeeScript features or adding new ones where missing. 
+**ES2015 features that modern frameworks require, like modules and classes, must be supported in CoffeeScript ASAP.** We can’t expect developers to continue using CoffeeScript if they must choose between CoffeeScript and whatever hot new framework they want to use for their project. Support for modules is [in progress](https://github.com/GeoffreyBooth/coffeescript/pull/2).
```
Comment (rattrayalex) [inline README.md]: I don't love using bold for emphasis, especially for a whole sentence. What about giving that sentence its own paragraph break?

**Comment by rattrayalex** [comment]
Thanks @GeoffreyBooth ! I like what I see. 

It seems a bit long for a README. Thoughts on how we could shorten it? Would you like me to try to help edit it down? Also happy to merge as-is if you think that's best.

**Comment by GeoffreyBooth** [comment]
Why don’t we merge it and then we can discuss what to cut or split out into separate files. On the one hand I feel like the proposal could be its own file, but on the other I feel like that’s the central thing in this repo that we want people to notice and comment on, so it should pop up on the first page. I dunno.

One thing we should maybe create as a separate top-level markdown file is a list of what we’ve agreed on. For example it seems like we’ve reached a consensus on `const`, so it would be good to put the plan in writing so it’s clearer than a stream of issue comments. Maybe the “way forward” section could become part of that file, and that’s a living document where we hash out the specifics of how we’re going to achieve our goals.

File: README.md (hunk 2 above)
Comment (GeoffreyBooth) [inline README.md]: Good point, I’ll revise this the next time I edit the document. Or you can submit a PR 😉

**Comment by rattrayalex** [comment]
> I wanted to submit this as a PR rather than just overwriting the README

We should probably always send as PR for anything other than a typo fix, for transparency.

**Comment by rattrayalex** [comment]
I'd rather not to a sloppy merge into the README; if we like this as it is, cool, but if we think we can do better I'd rather that happen before merging.

**Comment by GeoffreyBooth** [comment]
How about the “A New Compiler” and ”Relevant Discussions” sections get split into their own files? “A New Compiler” is very specific about the technical plan, which doesn’t need to be in a top-level README; and “Relevant Discussions” is becoming less relevant now that the coffeescript6 repo itself exists, and our discussions (in my opinion) supersede any older ones from the coffeescript main repo.

**Comment by rattrayalex** [comment]
Nice, that sounds good to me!

**Comment by rattrayalex** [comment]
Nice, thanks! Probably can also move the "possible alternatives" to reference file as well. I can do in a follow-on if you prefer.

**Pull request merged**